hook.js

```diff
@@ -138,6 +154,41 @@ set.${n} = (v) => {
 `).join('\n')}
 register(${JSON.stringify(realUrl)}, namespace, set, ${JSON.stringify(specifiers.get(realUrl))})
 `
+      } 
+    }
+    else if (NODE_MAJOR >= 16 && context.format === 'module') {
+      let fileContents
+      try {
+        fileContents = fs.readFileSync(fileURLToPath(url), 'utf8')
+      } catch (parseError) {
+        console.error(`Had trouble reading file: ${fileContents}, got error: ${parseError}`)
```

Comment:
Don't use `console.error` in non-test code. If logging is handy here, use [`debuglog`](https://nodejs.org/dist/latest-v21.x/docs/api/util.html#utildebuglogsection-callback)

Reply:
You can also use `warn`/`process.emitWarning` as is done elsewhere.